### PR TITLE
Webpack 3 and other perks

### PR DESCRIPTION
**Pull request by diondirza**
This PR include:
- webpack 3 upgrade, include scope hoisting in our client build and optimize module loader
- remove unnecessary config value
- other deps upgrade
- fix security issue with font source being loaded from origin with prefix `data:`

### CODE REVIEW

**Comment by strues** [comment]
Runs great. Going to merge this and update the Apollo PR with it.

File: internal/webpack/configFactory.js [hunk 1]
```diff
@@ -212,6 +219,11 @@ export default function webpackConfigFactory(buildOptions) {
           }),
       ),
 
+      // Implement webpack 3 scope hoisting that will remove function wrappers
+      // around your modules you may see some small size improvements. However,
+      // the significant improvement will be how fast the JavaScript loads in the browser.
+      ifClient(new webpack.optimize.ModuleConcatenationPlugin()),
```
Comment (strues) [inline internal/webpack/configFactory.js]: I'd argue this should be used only in production builds since it impacts build speed. Maybe I'm wrong?

Comment (ctrlplusb) [inline internal/webpack/configFactory.js]: Agreed 😀

Comment (diondirza) [inline internal/webpack/configFactory.js]: new commit have fixed that, please review again guys.

**Pull request merged**